### PR TITLE
add detail desc for String Function SUBSTRING_INDEX () and TO_BASE64

### PR DESCRIPTION
**Pull request by lyssom**
Fix https://github.com/pingcap/docs-cn/issues/15950

### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [ ] master (the latest development version)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: functions-and-operators/string-functions.md
Comment (xzhangxian1008) [inline functions-and-operators/string-functions.md:1]: - 冒号请用`:`
- 中文和英文请用空格隔开

File: functions-and-operators/string-functions.md [hunk 1]
```diff
@@ -292,11 +292,34 @@ SELECT BIN("123q123");
 
 ### [`SUBSTRING_INDEX()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_substring-index)
 
-从一个字符串中返回指定出现次数的定界符之前的子字符串
+定义：返回字符串str中出现count次分隔符delim之前的子字符串。如果count为正数，则返回最终分隔符左侧的所有内容（从左侧算起）。
```
Comment (xzhangxian1008) [inline functions-and-operators/string-functions.md:295]:
```suggestion
定义: 返回字符串 str 中出现第 count 次分隔符 delim 之前的子字符串。如果 count 为正数，则返回最终分隔符左侧的所有内容（从左侧算起）。
```

File: functions-and-operators/string-functions.md [hunk 2]
```diff
@@ -292,11 +292,34 @@ SELECT BIN("123q123");
...
+      如果count为负数，则返回最终分隔符右侧（从右侧计数）的所有内容。
+      SUBSTRING_INDEX()在搜索delim时执行区分大小写的匹配
```
Comment (xzhangxian1008) [inline functions-and-operators/string-functions.md:297]:
```suggestion
      SUBSTRING_INDEX() 在搜索 delim 时执行大小写敏感的匹配
```

File: functions-and-operators/string-functions.md [hunk 3]
```diff
@@ -292,11 +292,34 @@ SELECT BIN("123q123");
...
+语法：SUBSTRING_INDEX(str,delim,count)
```
Comment (xzhangxian1008) [inline functions-and-operators/string-functions.md:298]: 参数和逗号之间请加空格
```suggestion
语法: SUBSTRING_INDEX(str, delim, count)
```

File: functions-and-operators/string-functions.md [hunk 4]
```diff
@@ -292,11 +292,34 @@ SELECT BIN("123q123");
...
+参数: str,delim,count。必须项。字符串，分隔符，个数。
+返回值：子字符串
+SELECT SUBSTRING_INDEX('www.tidb.com', '.', 2);
```
Comment (xzhangxian1008) [inline functions-and-operators/string-functions.md:301]: 请添加一个 count 为负数的例子

File: functions-and-operators/string-functions.md [hunk 5]
```diff
@@ -292,11 +292,34 @@ SELECT BIN("123q123");
 
 ### [`SUBSTRING_INDEX()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_substring-index)
 
-从一个字符串中返回指定出现次数的定界符之前的子字符串
+定义：返回字符串str中出现count次分隔符delim之前的子字符串。如果count为正数，则返回最终分隔符左侧的所有内容（从左侧算起）。
+      如果count为负数，则返回最终分隔符右侧（从右侧计数）的所有内容。
+      SUBSTRING_INDEX()在搜索delim时执行区分大小写的匹配
+语法：SUBSTRING_INDEX(str,delim,count)
+参数: str,delim,count。必须项。字符串，分隔符，个数。
+返回值：子字符串
```
Comment (xzhangxian1008) [inline functions-and-operators/string-functions.md:300]: 返回值有点迷惑，指的是返回类型吧？

File: functions-and-operators/string-functions.md [hunk 6]
```diff
@@ -292,11 +292,34 @@ SELECT BIN("123q123");
 
 ### [`SUBSTRING_INDEX()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_substring-index)
 
-从一个字符串中返回指定出现次数的定界符之前的子字符串
+定义：返回字符串str中出现count次分隔符delim之前的子字符串。如果count为正数，则返回最终分隔符左侧的所有内容（从左侧算起）。
+      如果count为负数，则返回最终分隔符右侧（从右侧计数）的所有内容。
```
Comment (xzhangxian1008) [inline functions-and-operators/string-functions.md:296]: 定义中无需换行

File: functions-and-operators/string-functions.md [hunk 7]
```diff
@@ -292,11 +292,34 @@ SELECT BIN("123q123");
...
+| SUBSTRING_INDEX('www.tidb.com', '.', 2) |
++-----------------------------------------+
+| www.tidb                                |
++-----------------------------------------+
 
 ### [`TO_BASE64()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_to-base64)
 
-返回转化为 base-64 表示的字符串参数
+定义：将字符串参数转换为base-64编码形式，并将结果作为具有连接字符集和排序规则的字符串返回。
+      如果参数不是字符串，则在转换之前将其转换为字符串。
```
Comment (xzhangxian1008) [inline functions-and-operators/string-functions.md:311]: 定义中无需换行

File: functions-and-operators/string-functions.md [hunk 8]
```diff
@@ -292,11 +292,34 @@ SELECT BIN("123q123");
...
++-----------------------------------------+
+| SUBSTRING_INDEX('www.tidb.com', '.', 2) |
++-----------------------------------------+
+| www.tidb                                |
++-----------------------------------------+
 
 ### [`TO_BASE64()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_to-base64)
 
-返回转化为 base-64 表示的字符串参数
+定义：将字符串参数转换为base-64编码形式，并将结果作为具有连接字符集和排序规则的字符串返回。
```
Comment (xzhangxian1008) [inline functions-and-operators/string-functions.md:310]:
```suggestion
定义: 将字符串参数转换为 base-64 编码形式，并将带有连接字符集和 collation 的字符串作为结果返回。
```
collation 无需翻译，直接写 collation 就行

File: functions-and-operators/string-functions.md [hunk 9]
```diff
@@ -292,11 +292,34 @@ SELECT BIN("123q123");
...
+      如果参数不是字符串，则在转换之前将其转换为字符串。
+      如果参数为NULL，则结果为NULL。
+      Base-64编码的字符串可以使用FROM_BASE64()函数进行解码。
+语法：TO_BASE64(str)
+参数: str,必须项。待编码的字符串。
+返回值：base-64编码
```
Comment (xzhangxian1008) [inline functions-and-operators/string-functions.md:316]: @qiancai 我们的文档中需要写返回值吗？

File: functions-and-operators/string-functions.md [hunk 10]
```diff
@@ -292,11 +292,34 @@ SELECT BIN("123q123");
...
 
 ### [`TO_BASE64()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_to-base64)
 
-返回转化为 base-64 表示的字符串参数
+定义：将字符串参数转换为base-64编码形式，并将结果作为具有连接字符集和排序规则的字符串返回。
+      如果参数不是字符串，则在转换之前将其转换为字符串。
+      如果参数为NULL，则结果为NULL。
+      Base-64编码的字符串可以使用FROM_BASE64()函数进行解码。
+语法：TO_BASE64(str)
+参数: str,必须项。待编码的字符串。
```
Comment (xzhangxian1008) [inline functions-and-operators/string-functions.md:315]: 参数后面的格式是否有问题 @qiancai

File: functions-and-operators/string-functions.md [hunk 11]
```diff
@@ -292,11 +292,34 @@ SELECT BIN("123q123");
...
+返回值：base-64编码
+SELECT TO_BASE64('abc');
```
Comment (xzhangxian1008) [inline functions-and-operators/string-functions.md:317]: 请在添加一个参数为非字符串的例子

File: functions-and-operators/string-functions.md
Comment (lyssom) [reply functions-and-operators/string-functions.md:1]: ok，I fixed it

File: functions-and-operators/string-functions.md [hunk 12]
```diff
@@ -292,11 +292,40 @@ SELECT BIN("123q123");
 
 ### [`SUBSTRING_INDEX()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_substring-index)
 
-从一个字符串中返回指定出现次数的定界符之前的子字符串
-
+定义:返回字符串 str 中出现 count 次分隔符 delim 之前的子字符串。如果 count 为正数，则返回最终分隔符左侧的所有内容（从左侧算起）。如果 count 为负数，则返回最终分隔符右侧（从右侧计数）的所有内容。SUBSTRING_INDEX() 在搜索 delim 时执行区分大小写的匹配
```
Comment (xzhangxian1008) [inline functions-and-operators/string-functions.md:295]: 冒号后面请添加空格，其它的地方同样。

Comment (lyssom) [reply functions-and-operators/string-functions.md]: ok

**Review by xzhangxian1008 — APPROVED** [review]
LGTM

**Comment by qiancai** [comment]
Hi @lyssom, I've submitted the commit https://github.com/pingcap/docs-cn/pull/16112/commits/92ef24fda0e22917e215cbfecae915d423118d45 to refine some descriptions and unify the formats. And I updated "www.tidb.com" to "www.tidbcloud.com" because the latter is a valid web address. Would you please check whether the commit is OK? Thanks.

**Review by qiancai — APPROVED** [review]

File: functions-and-operators/string-functions.md [hunk 13]
```diff
@@ -292,11 +292,97 @@ SELECT BIN("123q123");
...
+语法：
+
+```sql
+SUBSTRING_INDEX(str, delim, count)
+```
+
+- `str`：要处理的字符串。
+- `delim`：指定字符串中的分隔符，大小写敏感。
+- `count`：指定分隔符出现的次数。
+    - 如果 `count` 为正数，该函数返回返回从字符串左边开始的第 `count` 个分隔符之前的子字符串。
```
Comment (qiancai) [inline functions-and-operators/string-functions.md:306]:
```suggestion
    - 如果 `count` 为正数，该函数返回从字符串左边开始的第 `count` 个分隔符之前的子字符串。
```

File: functions-and-operators/string-functions.md [hunk 14]
```diff
@@ -292,11 +292,97 @@ SELECT BIN("123q123");
...
+| SUBSTRING_INDEX('www.tidbcloud.com', '.', -1) |
++------------------------------------------+
+| com                                      |
++------------------------------------------+
+```
 
 ### [`TO_BASE64()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_to-base64)
 
-返回转化为 base-64 表示的字符串参数
+`TO_BASE64()` 函数用于将输入的参数转换为 base-64 编码形式的字符串，并按照当前会话的字符集和排序规则返回结果。base-64 编码的字符串可以使用  [`FROM_BASE64()`](#frombase64httpsdevmysqlcomdocrefman80enstring-functionshtmlfunctionfrom-base64) 函数进行解码。
```
Comment (qiancai) [inline functions-and-operators/string-functions.md:344]:
```suggestion
`TO_BASE64()` 函数用于将输入的参数转换为 base-64 编码形式的字符串，并按照当前会话的字符集和排序规则返回结果。base-64 编码的字符串可以使用  [`FROM_BASE64()`](#from_base64) 函数进行解码。
```

File: functions-and-operators/string-functions.md [hunk 15]
```diff
@@ -292,11 +292,97 @@ SELECT BIN("123q123");
...
+| SUBSTRING_INDEX('www.tidbcloud.com', '.', -1) |
++------------------------------------------+
+| com                                      |
++------------------------------------------+
+```
 
 ### [`TO_BASE64()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_to-base64)
 
-返回转化为 base-64 表示的字符串参数
+`TO_BASE64()` 函数用于将输入的参数转换为 base-64 编码形式的字符串，并按照当前会话的字符集和排序规则返回结果。base-64 编码的字符串可以使用  [`FROM_BASE64()`](#from_base64) 函数进行解码。
```
Comment (qiancai) [inline functions-and-operators/string-functions.md:344]:
```suggestion
`TO_BASE64()` 函数用于将输入的参数转换为 base-64 编码形式的字符串，并按照当前会话的字符集和排序规则返回结果。base-64 编码的字符串可以使用 [`FROM_BASE64()`](#from_base64) 函数进行解码。
```

**Pull request merged**